src/Streamly/Coreutils/FileTest.hs

```diff
@@ -322,3 +361,67 @@ isOlderThan = compareModTime (<)
 -- Like coreutil @test file1 -ef file2@.
 isSameFile :: FilePath -> IO FileTest
 isSameFile = undefined
+
+data TimeStampMin = TimeStampMin Ordering Int
+
+data FileSize = FileSize Ordering Integer
+
+getLocalTime :: IO MilliSecond64
+getLocalTime = fromAbsTime <$> getTime Realtime
+
+compareFsTime :: (FileStatus -> EpochTime) -> TimeStampMin -> IO FileTest
+compareFsTime toEpoc fopt = do
+    t <- getLocalTime
+    return $ FileTest (Predicate ( `f` t))
+
+    where
+
+    f st t =
+        case toEpoc st of
+            CTime t1 ->
+                let t2 =
+                        case fopt of
+                            TimeStampMin LT n ->
+                                t <
+                                MilliSecond64
+                                    (t1 * 1000 + fromIntegral (n * 60000))
+                            TimeStampMin GT n ->
+                                t >
+                                MilliSecond64
+                                    (t1 * 1000 + fromIntegral (n * 60000))
+                            TimeStampMin EQ n ->
+                                t ==
+                                MilliSecond64
+                                    (t1 * 1000 + fromIntegral (n * 60000))
+                  in t2
+
+compareAccessTime :: TimeStampMin -> IO FileTest
+compareAccessTime = compareFsTime Files.accessTime
+
+compareChangeTime :: TimeStampMin -> IO FileTest
+compareChangeTime = compareFsTime Files.modificationTime
+
+{-
+combinedFileTest :: IO FileTest
+combinedFileTest = do
+    ft1 <- compareAccessTime $ TimeStampMin LT 5
+    ft2 <- compareChangeTime $ TimeStampMin LT 10
+    return $ ft1 <> ft2
+-}
+
+compareFileSize :: FileSize -> IO FileTest
```

Comment:
The signature should be `compareFileSize :: (Integer -> Integer -> Bool) -> FilePath -> IO FileTest`. Then we could write more specialized checks like `isBiggerThan` or `isSmallerThan`.